pkg/codegen/nodejs/gen_program.go

```diff
@@ -376,31 +376,71 @@ type programImports struct {
 func (g *generator) collectProgramImports(program *pcl.Program) programImports {
 	importSet := codegen.NewStringSet("@pulumi/pulumi")
 	preambleHelperMethods := codegen.NewStringSet()
-	var componentImports []string
 
+	// This map tracks the package tokens by the associated import.
+	//
+	// It will have entries similar to the following:
+	//
+	// npmToPuPkgName["@pulumiverse/scaleway"] = "scaleway"
 	npmToPuPkgName := make(map[string]string)
+
+	var componentImports []string
 	seenComponentImports := map[string]bool{}
+
+	// Index known PackageReference by corresponding package name token.
+	knownPackageRefsByPkg := make(map[string]schema.PackageReference)
+	for _, pkgRef := range program.PackageReferences() {
+		// Can more than one package be bound to the same name? It seems possible in case of components.
```

Comment:
This is the question. Looking at PackageReferences() definition it does not seem to be obvious they are unique by name.